### ISSUE TITLE
LivePush SecretPopulationError with v13.1.1 or newer

### ISSUE THREAD

**Issue by volks73** [balena-io/balena-cli#2440]
# Expected Behavior

I was trying to use the Balena LivePush feature with Balena CLI v13.1.1 and v13.1.7 from a Windows Subsystem for Linux (WSL) v2 with Ubuntu 20.04 environment and a native Linux Ubuntu 20.04 machine with a multi-container project that contains a `.balena/secrets/fury-deploy-token` and the following `balena.yml` file:

```yaml
build-secrets:
  services:
     backend:
       - source: fury-deploy-token
         dest: fury-deploy-token
```

The `fury-deploy-token` contains a single line with no trailing newline characters and no slashes.

I would expect the Balena LivePush feature to successfully build the images and work like it did with Balena CLI v13.1.0.

# Actual Behavior

When I run the `balena push 7871a2e.local --debug` command after using the `balena login` with web authentication, I get the following output:

```
[debug] Deprecation check: 0.03249 days since last npm registry query for next major version release date.
[debug] Will not query the registry again until at least 7 days have passed.
[Debug]   Using build source directory: .
[Debug]   Pushing to local device: 7871a2e.local
[Debug]   Checking we can access device
[Debug]   Sending request to http://192.168.1.107:48484/ping
[Debug]   Checking device supervisor version: 12.11.22
[Info]    Starting build on device 192.168.1.107
[Debug]   Loading project...
[Debug]   Resolving project...
[Debug]   docker-compose.yml file found at "."
[Debug]   Creating project...
[Debug]   Tarring all non-ignored files...
[Debug]   Tarring complete in 138 ms
[Debug]   Fetching device information...
[Debug]   Sending request to http://192.168.1.107:48484/v2/local/device-info
[Debug]   Found build tasks:
[Debug]       dashboard: build [./dashboard]
[Debug]       database: build [./database]
[Debug]       backend: build [./backend]
[Debug]   Resolving services with [jetson-xavier-nx-devkit|aarch64]
[Debug]   Found project types:
[Debug]       dashboard: Standard Dockerfile
[Debug]       database: Standard Dockerfile
[Debug]       backend: Dockerfile.template
[Debug]   Probing remote daemon for cache images
[Debug]   Using 136 on-device images for cache...
[Debug]   Starting builds...
(HTTP code 500) server error - invalid character '/' looking for beginning of value

SecretPopulationError: (HTTP code 500) server error - invalid character '/' looking for beginning of value
    at populateSecrets (/snapshot/versioned-source/node_modules/resin-multibuild/build/build-secrets/index.js:87:15)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)

For further help or support, visit:
https://www.balena.io/docs/reference/balena-cli/#support-faq-and-troubleshooting
```

Uninstalling v13.1.1 or newer and re-installing v13.1.0 will successfully complete LivePush.

# Steps to Reproduce the Problem

1. Install Balena v13.1.1 or newer
2. Create a `.balena/balena.yml` file
3. Create a `.balena/secrets/fury-deploy-token` file
4. Login to BalenaCloud, `balena login`
5. Change a device to local mode
6. Initiate LivePush, `balena push <SHORT-UUID>.local`, where `<SHORT-UUID>` is replaced with the short UUID of the device in local mode.

# Specifications

- **balena CLI version:** balena-cli version "13.1.7" and Node.js version "12.22.7"
- **balenaCloud**
- **Operating system version:** Windows 10 WSL2 Ubuntu 20.04 and Linux Ubuntu 20.04
- **64 bit OS and processor:** Windows 10 WSL2 Intel i7 and Linux Ubuntu 20.04 AMD Threadripper
- **Install method:** zip package
- **If npm install, Node.js and npm version:** Not applicable

# Additional References

I originally submitted a forum post when Balena CLI v13.1.1 was released: https://forums.balena.io/t/balena-cli-and-livepush-secretpopulationerror/350751, which may contain more helpful details. Seeing the several bug/hotfix releases since that forum post, I thought the error would be resolved, but the same error occurs for v13.1.7. 

The [Balena Secrets](https://www.balena.io/docs/learn/deploy/deployment/#build-time-only-secret-file) documentation was consulted for creating the secrets structure and `balena.yml` file.

**Comment by pdcastro** [comment]
Thank you for reporting this issue @volks73 👍 

@Page-, as this issue was introduced in v13.1.1, it would have been PR #2430. It needs some investigation to find out which of the updated dependencies would have caused it.

**Comment by Page-** [comment]
@pdcastro my first thoughts would be the dockerode/docker-modem updates since the error looks to me like it's coming from balena-engine (docker) on the device that is being live pushed to

**Comment by pdcastro** [comment]
@Page-, indeed I traced the issue to `docker-modem` v3.0.1, specifically this PR: https://github.com/apocas/docker-modem/pull/133#issuecomment-1009464228

A workaround is to pin `docker-modem` to v3.0.0. I think I will create a PR for this temporary pinning now, which would give @robertgzr and the `docker-modem` maintainer some time to decide on the best proper fix.

**Comment by pdcastro** [comment]
@volks73, this issue is fixed in balena CLI v13.1.8 or later. Thanks again for reporting it 👍

**Comment by volks73** [comment]
@pdcastro, your welcome. Thank you for the fix and effort to resolve the issue. It is greatly appreciated.
